src/adapter/src/optimize/peek.rs

```diff
@@ -343,7 +387,12 @@ impl GlobalMirPlan<Unresolved> {
         // we expect to be able to set `until = as_of + 1` without an overflow.
         if let Some(as_of) = timestamp_ctx.timestamp() {
```

Comment:
@mjibson can I understand that `SELECT` can happen in a `TimestampContext::NoTimestamp` context, but do you think that we should allow this for `COPY TO`. If yes - what should be the timestamp for the configured S3 sink?

Reply:
It should be possible to write `COPY (SELECT 1) TO ...` which dumps that one row to s3. We actually intended to use that as a quick check that the aws connection + perms are set up correctly.

Reply:
Then maybe some of the changes in this this bit of the code that I'm proposing in `optimize/copy_to.rs` in the other PR should be reverted (return the plan and not a `Result`), don't return an `OptimizerError::Internal` when the timestamp is not present.

Reply:
Yup, will remove that bit 👍

Reply:
(done, and added a test)